zhenggen-xu/sonic-utilities
# [config_mgmt.py]: Shutdown the interfaces after config validation.

Changes:
-- Shutdown the interfaces after config validation while Dy Port Breakout.
-- lib.py: to avoid loop dependencies b/w config_mgmt.py and main.py
   moved few helper funtions in lib.py.

Signed-off-by: Praveen Chaudhary pchaudhary@linkedin.com



**- What I did**
Changes:
-- Shutdown the interfaces after config validation while Dy Port Breakout.
-- lib.py: to avoid loop dependencies b/w config_mgmt.py and main.py
   moved few helper funtions in lib.py.

**- How I did it**
-- move few helper functions to lib.py.
-- move shutdown part to config_mgmt.py

**- How to verify it**
```
svc-lnos-user@server10:/home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests$ cat test_port_dpb_system.py | grep -A 4 -B 4 Uncomment

        # Verify port is removed from ACL table
        self.dvs_acl.verify_acl_table_count(1)

        #TBD: Uncomment this, or explain why Ethernet0 is being added back to ACL table
        # Also, string "None" is being added as port to ACL port list after the breakout
        self.dvs_acl.verify_acl_group_num(0)

        # Verify child ports are created.

svc-lnos-user@server10:/home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests$ sudo pytest --dvsname=vs-jk test_port_dpb_system.py -k test_port_breakout_with_acl --junitxml=report.xml | tee out
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-3.3.0, py-1.6.0, pluggy-0.6.0
rootdir: /home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 2 items

test_port_dpb_system.py .                                                [100%]

 generated xml file: /home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests/report.xml
============================== 1 tests deselected ==============================
=================== 1 passed, 1 deselected in 92.36 seconds ====================
```

```
Jun  5 06:16:37.007136 fec26aa9f1be INFO #sonic_yang: Try to load Data in the tree
Jun  5 06:16:37.007722 fec26aa9f1be INFO #configMgmt: Data Validation successful
Jun  5 06:16:37.031857 fec26aa9f1be INFO #configMgmt: shutdown Interfaces: {'PORT': {'Ethernet2': {'admin_status': 'down'}, 'Ethernet3': {'admin_status': 'down'}, 'Ethernet0': {'admin_status': 'down'}, 'Ethernet1': {'admin_status': 'down'}}}
Jun  5 06:16:37.032485 fec26aa9f1be INFO #configMgmt: Write in DB: {'PORT': OrderedDict([('Ethernet0', {'admin_status': 'down'}), ('Ethernet1', {'admin_status': 'down'}), ('Ethernet2', {'admin_status': 'down'}), ('Ethernet3', {'admin_status': 'down'})])}
Jun  5 06:16:37.033823 fec26aa9f1be NOTICE #orchagent: :- doPortTask: Set port Ethernet0 admin status to down
Jun  5 06:16:37.034270 fec26aa9f1be INFO #configMgmt: Write in DB: {u'PORT': OrderedDict([(u'Ethernet0', None), (u'Ethernet1', None), (u'Ethernet2', None), (u'Ethernet3', None)])}
Jun  5 06:16:37.034383 fec26aa9f1be NOTICE #orchagent: :- doPortTask: Set port Ethernet1 admin status to down
Jun  5 06:16:37.034486 fec26aa9f1be NOTICE #orchagent: :- doPortTask: Set port Ethernet2 admin status to down
Jun  5 06:16:37.034872 fec26aa9f1be NOTICE #portsyncd: :- onMsg: nlmsg type:16 key:Ethernet0 admin:0 oper:0 addr:02:42:ac:11:00:0c ifindex:1199 master:0 type:tun
Jun  5 06:16:37.035004 fec26aa9f1be NOTICE #orchagent: :- doPortTask: Set port Ethernet3 admin status to down
Jun  5 06:16:37.035166 fec26aa9f1be NOTICE #portsyncd: :- onMsg: Publish Ethernet0(ok) to state db
Jun  5 06:16:37.035317 fec26aa9f1be INFO #configMgmt: Verify Port Deletion from Asic DB, Wait...
```

——Test 06/09/2020
```
Jun  9 23:58:26.590188 fec26aa9f1be INFO #configMgmt: shutdown Interfaces: {'PORT': {'Ethernet2': {'admin_status': 'down'}, 'Ethernet0': {'admin_status': 'down'}}}
Jun  9 23:58:26.590828 fec26aa9f1be INFO #configMgmt: Write in DB: {'PORT': OrderedDict([('Ethernet0', {'admin_status': 'down'}), ('Ethernet2', {'admin_status': 'down'})])}
Jun  9 23:58:26.591800 fec26aa9f1be INFO #configMgmt: Write in DB: {u'PORT': OrderedDict([(u'Ethernet0', None), (u'Ethernet2', None)])}
Jun  9 23:58:26.592121 fec26aa9f1be INFO #configMgmt: Verify Port Deletion from Asic DB, Wait...

svc-lnos-user@server10:/home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests$ cat test_port_dpb_system.py | grep -A 4 -B 4 Uncomment

        # Verify port is removed from ACL table
        self.dvs_acl.verify_acl_table_count(1)

        #TBD: Uncomment this, or explain why Ethernet0 is being added back to ACL table
        # Also, string "None" is being added as port to ACL port list after the breakout
        self.dvs_acl.verify_acl_group_num(0)

        # Verify child ports are created.
svc-lnos-user@server10:/home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests$ sudo pytest --dvsname=vs-jk test_port_dpb_system.py -k test_port_breakout_with_acl --junitxml=report.xml | tee out
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-3.3.0, py-1.6.0, pluggy-0.6.0
rootdir: /home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 2 items

test_port_dpb_system.py .                                                [100%]

 generated xml file: /home/svc-lnos-user/praveen/sonic-buildimage/src/sonic-swss/tests/report.xml
============================== 1 tests deselected ==============================
=================== 1 passed, 1 deselected in 93.69 seconds ====================
```
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
